### PR TITLE
some fixes for BLSTM

### PR DESCRIPTION
**Pull request by freewym**

### CODE REVIEW

**Comment by danpovey** [comment]
Thanks-- Vijay, I'll wait for you to check this.

File: egs/swbd/s5c/RESULTS [hunk 1]
```diff
@@ -105,23 +105,17 @@ exit 0
...
 # and volume perturbed data)
-%WER 15.20 [ 7477 / 49204, 1031 ins, 1912 del, 4534 sub ] exp/nnet3/lstm_1280_384_123_ld5_sp/decode_train_dev_sw1_fsh_fg/wer_12_0.0
-%WER 16.29 [ 8013 / 49204, 1121 ins, 1908 del, 4984 sub ] exp/nnet3/lstm_1280_384_123_ld5_sp/decode_train_dev_sw1_tg/wer_11_0.0
-%WER 11.4 | 1831 21395 | 89.8 6.8 3.4 1.2 11.4 46.0 | exp/nnet3/lstm_1280_384_123_ld5_sp/decode_eval2000_sw1_fsh_fg/score_11_0.0/eval2000_hires.ctm.swbd.filt.sys
-%WER 12.7 | 1831 21395 | 88.6 7.6 3.8 1.3 12.7 48.7 | exp/nnet3/lstm_1280_384_123_ld5_sp/decode_eval2000_sw1_tg/score_11_0.0/eval2000_hires.ctm.swbd.filt.sys
+%WER 11.4 | 1831 21395 | 89.8 6.8 3.4 1.2 11.4 46.0 | exp/nnet3/lstm_ld5_sp/decode_eval2000_sw1_fsh_fg/score_11_0.0/eval2000_hires.ctm.swbd.filt.sys
+%WER 12.7 | 1831 21395 | 88.6 7.6 3.8 1.3 12.7 48.7 | exp/nnet3/lstm_ld5_sp/decode_eval2000_sw1_tg/score_11_0.0/eval2000_hires.ctm.swbd.filt.sys
 
+# local/nnet3/run_lstm.sh --affix bidirectional --lstm-delay " [-1,1] [-2,2] [-3,3] " --label-delay 0 --cell-dim 1024 --recurrent-projection-dim 128 --non-recurrent-projection-dim 128 --chunk-right-context 40
```
Comment (vijayaditya) [inline egs/swbd/s5c/RESULTS]: Modify the lines for readability
<code><pre>
\# bidirectional LSTM
\#--------------------------
\# local/nnet3/run_lstm.sh --affix bidirectional \
\#                                       --lstm-delay " [-1,1] [-2,2] [-3,3] " \
\#                                       --label-delay 0 \
\#                                       --cell-dim 1024 \
\#                                       --recurrent-projection-dim 128 \
\#                                       --non-recurrent-projection-dim 128 \
\#                                       --chunk-right-context 40
\#  Trained on (8 epoch training on speed-perturbed and volume perturbed data)
\# num_params=20101172 
</pre></code>



BTW also specify chunk-left-context in the parameters.

File: egs/swbd/s5c/run.sh [hunk 2]
```diff
@@ -301,5 +301,11 @@ fi
 # demonstration script for raw-fMLLR.  You should probably ignore this.
 # local/run_raw_fmllr.sh
 
+# nnet3 LSTM recipe
+# local/nnet3/run_lstm.sh
+
+# nnet3 BLSTM recipe
+# local/nnet3/run_lstm.sh --affix bidirectional --lstm-delay " [-1,1] [-2,2] [-3,3] " --label-delay 0 --cell-dim 1024 --recurrent-projection-dim 128 --non-recurrent-projection-dim 128 --chunk-right-context 40
```
Comment (vijayaditya) [inline egs/swbd/s5c/run.sh]: It might be better to modify this command to have one parameter per line.

File: egs/wsj/s5/RESULTS [hunk 3]
```diff
@@ -289,14 +289,16 @@ for x in exp/nnet3/nnet_tdnn_a/decode_*; do grep WER $x/wer_* | utils/best_wer.s
...
 %WER 6.40 [ 361 / 5643, 87 ins, 16 del, 258 sub ] exp/nnet3/nnet_tdnn_a/decode_tgpr_eval92/wer_10_1.0
 
+# local/nnet3/run_lstm.sh
 # LSTM results: cell_dim=1024, recurrent_projection_dim=non_recurrent_projection_dim=256,lstm_delay=-1 -2 -3, label_delay=5
 %WER 7.32 exp/nnet3/lstm_ld5/decode_bd_tgpr_dev93/wer_11_0.0
 %WER 4.24 exp/nnet3/lstm_ld5/decode_bd_tgpr_eval92/wer_10_1.0
 %WER 9.57 exp/nnet3/lstm_ld5/decode_tgpr_dev93/wer_9_1.0
 %WER 6.86 exp/nnet3/lstm_ld5/decode_tgpr_eval92/wer_10_1.0
 
+# local/nnet3/run_lstm.sh --affix bidirectional --lstm-delay " [-1,1] [-2,2] [-3,3] " --label-delay 0 --cell-dim 640 --recurrent-projection-dim 128 --non-recurrent-projection-dim 128 --chunk-right-context 40
```
Comment (vijayaditya) [inline egs/wsj/s5/RESULTS]: modify as shown in previous comment.

**Comment by freewym** [comment]
OK. Done.

**Comment by vijayaditya** [comment]
@danpovey good to merge.

**Comment by danpovey** [comment]
OK, merging.

**Pull request merged**